src/formats/notion-api/api-helpers.ts

```diff
@@ -0,0 +1,337 @@
+			() => client.blocks.children.list({
+				block_id: blockId,
+				start_cursor: cursor,
+				page_size: 100,
+			}),
+			ctx
+		);
+		
+		// Filter out partial blocks
+		const fullBlocks = response.results.filter(
+			(block: any): block is BlockObjectResponse => 'type' in block
+		);
+		
+		blocks.push(...fullBlocks);
+		cursor = response.has_more ? response.next_cursor ?? undefined : undefined;
+		
+	} while (cursor);
+	
+	return blocks;
+}
+
+/**
+ * Extract page title from Notion page object
+ */
+export function extractPageTitle(page: PageObjectResponse): string {
+	const properties = page.properties;
+	
+	// Try to find title property
+	for (const key in properties) {
+		const prop = properties[key];
+		if (prop.type === 'title' && prop.title.length > 0) {
+			return prop.title.map(t => t.plain_text).join('');
+		}
+	}
+	
+	return 'Untitled';
+}
+
+/**
+ * Extract frontmatter from Notion page
+ * This includes the notion-id, timestamps, and all page properties
+ * @param page - The Notion page object
+ * @param formulaStrategy - How to handle formula properties ('static', 'function', 'hybrid')
+ * @param databaseProperties - Database property schema (for checking if formulas can be converted)
+ */
+export function extractFrontMatter(
+	page: PageObjectResponse, 
+	formulaStrategy: 'static' | 'function' | 'hybrid' = 'function',
+	databaseProperties?: any
+): Record<string, any> {
+	const frontMatter: Record<string, any> = {
+		'notion-id': page.id,
+	};
+	
+	// Add created and last edited time
+	if (page.created_time) {
+		frontMatter.created = page.created_time;
+	}
+	if (page.last_edited_time) {
+		frontMatter.updated = page.last_edited_time;
+	}
+	
+	// Extract all page properties
+	const properties = page.properties;
+	for (const key in properties) {
+		const prop = properties[key];
+		
+		// Skip title property (already used as filename)
+		if (prop.type === 'title') {
+			continue;
+		}
+		
+		// Handle formula properties based on strategy
+		if (prop.type === 'formula') {
+			const shouldAddToYAML = shouldAddFormulaToYAML(
+				key,
+				databaseProperties,
+				formulaStrategy
+			);
+			
+			if (shouldAddToYAML) {
+				const value = mapNotionPropertyToFrontmatter(prop);
+				if (value !== null && value !== undefined) {
+					frontMatter[key] = value;
+				}
+			}
+			continue;
+		}
+		
+		// Map property to frontmatter value
+		const value = mapNotionPropertyToFrontmatter(prop);
+		if (value !== null && value !== undefined) {
+			frontMatter[key] = value;
+		}
+	}
+	
+	return frontMatter;
+}
+
+/**
+ * Determine if a formula property should be added to page YAML
+ * Based on the import strategy and whether the formula can be converted
+ */
+function shouldAddFormulaToYAML(
+	propertyKey: string,
+	databaseProperties: any,
+	strategy: 'static' | 'function' | 'hybrid'
+): boolean {
+	// Import formula-converter functions
+	const { canConvertFormula, getNotionFormulaExpression } = require('./formula-converter');
```

Comment:
Please move these imports to the top of the file and use import instead of require.